### PR TITLE
azure-messaging-webpubsub-client, address SDK arch review feedback

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Address feedback in https://apiview.dev/Assemblies/Review/c8eb7bc201f24a1b94bc424472916d49

considering adding a perf test, and check whether to switch to azure-json for stream-style seralization.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java
Comment (weidongxu-microsoft) [inline sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java:1]: package private class

**Review by zackliu — APPROVED** [review]

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java
Comment (srnagar) [reply sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java:1]: Do we need to keep this class? We can delete it if it's not going to be public.

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubClient.java [hunk 1]
```diff
@@ -56,7 +57,7 @@
  * <!-- end com.azure.messaging.webpubsub.client.WebPubSubClient -->
  */
 @ServiceClient(builder = WebPubSubClientBuilder.class)
-public class WebPubSubClient {
+public class WebPubSubClient implements Closeable {
```
Comment (srnagar) [inline sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubClient.java:60]: Make this `final`.

**Review by srnagar — APPROVED** [review]

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubClient.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubClient.java]: Thanks for pointing out. Made all user-facing class `final`.

The `WebPubSubAsyncClient` is invoked by this `WebPubSubClient` (it is Sync over Async under the hood).

`WebPubSubAsyncClient` is currently package private, so user should not see it.

**Pull request merged**